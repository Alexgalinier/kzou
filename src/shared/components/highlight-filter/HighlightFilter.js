import React from 'react';

export default ({ value, filter }) => {
  if (!filter) return value;

  let parts = value.split(new RegExp('(' + filter + ')', 'gi'));
  for (let i = 1; i < parts.length; i += 2) {
    parts[i] = (
      <span className="filter-match _bg-blue-1" key={i}>
        {parts[i]}
      </span>
    );
  }
  return <div>{parts}</div>;
};
