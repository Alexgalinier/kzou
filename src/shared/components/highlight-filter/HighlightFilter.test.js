import React from 'react';
import HighlightFilter from './HighlightFilter';
import renderer from 'react-test-renderer';

test('Highlight matching parts corresponding to filter in value', () => {
  const component = renderer.create(
    <HighlightFilter value="Denis Villeneuve" filter="enis" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Highlight nothing if not matching', () => {
  const component = renderer.create(
    <HighlightFilter value="Sidney Lumet" filter="enis" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Highlight multiple match', () => {
  const component = renderer.create(
    <HighlightFilter value="Martin Scorsese" filter="e" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Highlight case insensitive', () => {
  const component = renderer.create(
    <HighlightFilter value="Martin Scorsese" filter="E" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

