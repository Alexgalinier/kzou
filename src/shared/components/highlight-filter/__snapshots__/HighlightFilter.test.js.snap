// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight case insensitive 1`] = `
<div>
  Martin Scors
  <span
    className="filter-match _bg-blue-1"
  >
    e
  </span>
  s
  <span
    className="filter-match _bg-blue-1"
  >
    e
  </span>
  
</div>
`;

exports[`Highlight matching parts corresponding to filter in value 1`] = `
<div>
  D
  <span
    className="filter-match _bg-blue-1"
  >
    enis
  </span>
   Villeneuve
</div>
`;

exports[`Highlight multiple match 1`] = `
<div>
  Martin Scors
  <span
    className="filter-match _bg-blue-1"
  >
    e
  </span>
  s
  <span
    className="filter-match _bg-blue-1"
  >
    e
  </span>
  
</div>
`;

exports[`Highlight nothing if not matching 1`] = `
<div>
  Sidney Lumet
</div>
`;
